moj_analytics/auth0_client.py

```diff
@@ -104,26 +104,29 @@ def create(self, resource):
     def get_all(self, resource_class, **kwargs):
         endpoint = '{}s'.format(resource_class.__name__.lower())
 
-        resources = self.request('GET', endpoint, params=kwargs)
+        defaults = {
+            'page': 0,
+            'per_page': 100,
+            'include_totals': True
+        }
 
-        if endpoint in resources:
-            resources = resources[endpoint]
+        params = defaults.copy()
+        params.update(kwargs)
 
-        return [resource_class(self, r) for r in resources]
+        resources = []
 
-    def get_all_pages(self, resource_class, per_page=100, **kwargs):
-        page = 0
         while True:
-            items = self.get_all(
-                resource_class, per_page=per_page, page=page, **kwargs)
+            response = self.request('GET', endpoint, params=params)
```

Comment:
Hope to not sound annoying, but how would this behave if there are errors?

Presumably if the server doesn't respond `requests` raise an exception (if that happen halfway through that bubbles up which is good).

If status code is not `200 OK`, probably response doesn't have `endpoint` key, this is handled by skipping the page. Wouldn't this cause the loop to never end?

Reply:
Good point - request does not raise an exception for non-200 responses.